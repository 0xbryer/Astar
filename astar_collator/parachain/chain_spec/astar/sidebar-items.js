initSidebarItems({"fn":[["get_chain_spec","Gen Astar chain specification for given parachain id."]],"type":[["AstarChainSpec","Specialized `ChainSpec` for Astar Network."]]});