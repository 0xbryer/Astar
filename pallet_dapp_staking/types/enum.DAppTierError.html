<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/enum.DAppTierError.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/enum.DAppTierError.html">../../pallet_dapp_staking/enum.DAppTierError.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/enum.DAppTierError.html" + location.search + location.hash);</script>
</body>
</html>