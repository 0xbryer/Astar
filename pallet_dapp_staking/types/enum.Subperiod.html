<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/enum.Subperiod.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/enum.Subperiod.html">../../pallet_dapp_staking/enum.Subperiod.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/enum.Subperiod.html" + location.search + location.hash);</script>
</body>
</html>