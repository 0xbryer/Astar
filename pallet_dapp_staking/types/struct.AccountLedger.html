<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/struct.AccountLedger.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/struct.AccountLedger.html">../../pallet_dapp_staking/struct.AccountLedger.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/struct.AccountLedger.html" + location.search + location.hash);</script>
</body>
</html>