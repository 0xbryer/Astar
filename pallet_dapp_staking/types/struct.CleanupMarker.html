<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/struct.CleanupMarker.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/struct.CleanupMarker.html">../../pallet_dapp_staking/struct.CleanupMarker.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/struct.CleanupMarker.html" + location.search + location.hash);</script>
</body>
</html>