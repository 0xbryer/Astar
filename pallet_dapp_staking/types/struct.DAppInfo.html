<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/struct.DAppInfo.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/struct.DAppInfo.html">../../pallet_dapp_staking/struct.DAppInfo.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/struct.DAppInfo.html" + location.search + location.hash);</script>
</body>
</html>