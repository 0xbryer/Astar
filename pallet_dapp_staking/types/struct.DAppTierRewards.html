<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/struct.DAppTierRewards.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/struct.DAppTierRewards.html">../../pallet_dapp_staking/struct.DAppTierRewards.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/struct.DAppTierRewards.html" + location.search + location.hash);</script>
</body>
</html>