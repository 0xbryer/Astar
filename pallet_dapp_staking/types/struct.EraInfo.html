<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../pallet_dapp_staking/struct.EraInfo.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../pallet_dapp_staking/struct.EraInfo.html">../../pallet_dapp_staking/struct.EraInfo.html</a>...</p>
    <script>location.replace("../../pallet_dapp_staking/struct.EraInfo.html" + location.search + location.hash);</script>
</body>
</html>