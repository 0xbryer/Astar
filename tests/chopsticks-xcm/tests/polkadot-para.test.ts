import { Context } from '../networks/types'
import { query, tx } from '../helpers/api'

import { acala } from '../networks/acala'
import { hydraDX } from '../networks/hydraDX'
import { moonbeam } from '../networks/moonbeam'
import { statemint } from '../networks/statemint'
import { astar } from '../networks/astar'

import buildTest from './shared'


const tests = [
  // statemint <-> astar
  {
    from: 'statemint',
    to: 'astar',
    name: 'USDT',
    fromStorage: ({ alice }: Context) => ({
      System: {
        account: [[[astar.paraAccount], { data: { free: 10e10 } }]],
      },
      Assets: {
        account: [[[statemint.usdtIndex, alice.address], { balance: 1e8 }]],
        asset: [[[statemint.usdtIndex], { supply: 1e8 }]],
      },
    }),
    test: {
      xcmPalletHorizontal: {
        tx: tx.xcmPallet.limitedReserveTransferAssetsV3(statemint.usdt, 1e7, tx.xcmPallet.parachainV3(1, astar.paraId)),
        fromBalance: query.assets(statemint.usdtIndex),
        toBalance: query.assets(astar.usdt),
      },
    },
  },
] as const

export type TestType = (typeof tests)[number]

buildTest(tests)
