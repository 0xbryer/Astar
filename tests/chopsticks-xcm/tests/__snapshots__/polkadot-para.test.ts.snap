// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`'statemint' -> 'astar' xcm transfer 'USDT' > xcmPallet transfer > balance on from chain 1`] = `
{
  "balance": 90000000,
  "extra": null,
  "isFrozen": false,
  "reason": {
    "consumer": null,
  },
}
`;

exports[`'statemint' -> 'astar' xcm transfer 'USDT' > xcmPallet transfer > balance on to chain 1`] = `
{
  "balance": "(rounded 10000000)",
  "extra": null,
  "isFrozen": false,
  "reason": {
    "sufficient": null,
  },
}
`;

exports[`'statemint' -> 'astar' xcm transfer 'USDT' > xcmPallet transfer > from chain hrmp messages 1`] = `
[
  {
    "data": [
      "ConcatenatedVersionedXcm",
      {
        "v3": [
          {
            "reserveAssetDeposited": [
              {
                "fun": {
                  "fungible": 10000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
            ],
          },
          {
            "clearOrigin": null,
          },
          {
            "buyExecution": {
              "fees": {
                "fun": {
                  "fungible": 10000000,
                },
                "id": {
                  "concrete": {
                    "interior": {
                      "x3": [
                        {
                          "parachain": 1000,
                        },
                        {
                          "palletInstance": 50,
                        },
                        {
                          "generalIndex": 1984,
                        },
                      ],
                    },
                    "parents": 1,
                  },
                },
              },
              "weightLimit": {
                "unlimited": null,
              },
            },
          },
          {
            "depositAsset": {
              "assets": {
                "wild": {
                  "allCounted": 1,
                },
              },
              "beneficiary": {
                "interior": {
                  "x1": {
                    "accountId32": {
                      "id": "0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee",
                      "network": null,
                    },
                  },
                },
                "parents": 0,
              },
            },
          },
        ],
      },
    ],
    "recipient": 2006,
  },
]
`;

exports[`'statemint' -> 'astar' xcm transfer 'USDT' > xcmPallet transfer > to chain xcm events 1`] = `
[
  {
    "data": {
      "messageHash": "(hash)",
      "weight": {
        "proofSize": "(rounded 4100)",
        "refTime": 4000000000,
      },
    },
    "method": "Success",
    "section": "xcmpQueue",
  },
]
`;

exports[`'statemint' -> 'astar' xcm transfer 'USDT' > xcmPallet transfer > tx events 1`] = `
[
  {
    "data": [
      {
        "Complete": {
          "proofSize": "(rounded 6200)",
          "refTime": "(rounded 674000000)",
        },
      },
    ],
    "method": "Attempted",
    "section": "polkadotXcm",
  },
]
`;
